spec.md

```diff
@@ -1586,12 +1760,192 @@ en <= Z_valid
 cover(clk, pred, en, "X equals Y when Z is valid") : optional_name
 ```
 
+## Probes
+
+Probe references are created with `probe`{.firrtl} expressions, exported with
+the `export`{.firrtl} statement, forwarded between instances using the
+`forward`{.firrtl} statement, read using the `read`{.firrtl} expression (see
+[@sec:reading-probe-references]), and used with `force`{.firrtl} and
+`release`{.firrtl} statements.
+
+Export and forward are used to route references through the design, and may be
+used wherever is most convenient in terms of available identifiers.
+Every sink-flow probe must be the target of exactly one of these statements.
+
+These statements are detailed below.
+
+### Export
+
+The export statement takes a reference expression, which must be a probe expression,
+and exports the probe reference to into a sink-flow static reference target.
+
+```firrtl
+module Refs:
+  input clock:  Clock
+  output a : Probe<{x: UInt, y: UInt}> ; read-only ref. to wire 'p'
+  output b : RWProbe<UInt> ; force-able ref. to node 'q', inferred width.
+  output c : Probe<UInt<1>> ; read-only ref. to register 'r'
+  output d : RWProbe<UInt<2>>[4] ; vector of ref.'s to memory data in 'm'
+  output e : Probe<Clock> ; ref. to input clock port
+
+  wire p : {x: UInt, flip y : UInt}
+  export probe(p) as a ; probe is passive
+  node q = UInt<1>(0)
+  export rwprobe(q) as b
+  reg r: UInt, clock
+  export probe(r) as c
+  mem m:
+    data-type => UInt<5>
+    depth => 4
+    ; ...
+    read-under-write => undefined
+
+  export rwprobe(m) as d
+  export probe(clock) as e
+```
+
+Exporting to a field within a bundle or other statically known sub-element of
+an aggregate is allowed, for example:
+```firrtl
+module Foo:
+  input x : UInt
+  output y : {x: UInt, p: Probe<UInt>}
+  output z : Probe<UInt>[2]
+
+  wire p : UInt
+  p <= x
+  y.x <= p
+
+  export probe(p) as y.p
+  export probe(p) as z[0]
+  export probe(p) as z[1]
+```
+
+`RWProbe`{.firrtl} references to ports are not allowed on public-facing modules.
+
+#### Probes and Passive Types
+
+While `Probe`{.firrtl} inner types are passive, the type of the exported expression
+is not required to be:
+
+```firrtl
+module Foo :
+  input x : {a: UInt, flip b: UInt}
+  output y : {a: UInt, flip b: UInt}
+  output xp : Probe<{a: UInt, b: UInt}> ; passive
+
+  wire p : {a: UInt, flip b: UInt} ; p is not passive
+  export probe(p) as xp
+  p <= x
+  y <= p
+```
+
+#### Exporting References to Nested Declarations
+
+Nested declarations (see [@sec:nested-declarations]) may be exported:
+
+```firrtl
+module RefProducer :
+  input a : UInt
+  input en : UInt<1>
+  output thereg : Probe<UInt>
+
+  when en :
+    reg myreg : UInt, clk
+    myreg <= a
+    export probe(myreg) as thereg
```

Comment:
i dont get what this example is trying to show. what does this mean when `en` is false?

Reply:
It is equivalent to putting it elsewhere where that's legal, here that is not as `myreg` is only in scope within the `when` so that's the only place such can be written (yet we want this to be unconditional, aka independent of `en`).